numpy/core/src/common/ucsnarrow.c

```diff
@@ -107,10 +107,11 @@ PyUCS2Buffer_AsUCS4(Py_UNICODE *ucs2, npy_ucs4 *ucs4, int ucs2len, int ucs4len)
  * new_reference: PyUnicodeObject
  */
 NPY_NO_EXPORT PyUnicodeObject *
-PyUnicode_FromUCS4(char *src, Py_ssize_t size, int swap, int align)
+PyUnicode_FromUCS4(char const *src, Py_ssize_t size, int swap, int align)
 {
     Py_ssize_t ucs4len = size / sizeof(npy_ucs4);
-    npy_ucs4 *buf = (npy_ucs4 *)src;
+    /* FIXME: This is safe, but better to rewrite to not cast away const */
```

Comment:
Didn't want to do this rewrite as part of this patch, as it would be hidden amongst the noise